client/trino-cli/src/main/java/io/trino/cli/Trino.java

```diff
@@ -34,12 +46,31 @@ public static void main(String[] args)
 
     public static CommandLine createCommandLine(Object command)
     {
-        return new CommandLine(command)
+        CommandLine commandLine = new CommandLine(command)
                 .registerConverter(ClientResourceEstimate.class, ClientResourceEstimate::new)
                 .registerConverter(ClientSessionProperty.class, ClientSessionProperty::new)
                 .registerConverter(ClientExtraCredential.class, ClientExtraCredential::new)
                 .registerConverter(HostAndPort.class, HostAndPort::fromString)
                 .registerConverter(Duration.class, Duration::valueOf);
+
+        getConfigFile().ifPresent(file -> commandLine.setDefaultValueProvider(new CommandLine.PropertiesDefaultProvider(file)));
```

Comment:
Can we have unit tests for that? What if there are outdated or unrecognized properties?

Reply:
They are omitted. The same goes for the changed CLI options (changing name will be a breaking change here as well)

Reply:
CLI will fail for unrecognized option, it would be nice to fail as well when config file has unrecognized properties.

Reply:
@kokosing done, please review :)